From d45c8516d5efcc6ad33d01c672ac05e8e72556e3 Mon Sep 17 00:00:00 2001
From: Zhymabek Roman <robanokssamit@yandex.ru>
Date: Sat, 29 Oct 2022 16:40:39 +0600
Subject: [PATCH] Fix for noexec FS

---
 transcrypt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/transcrypt b/transcrypt
index 78fb2ce..1a2d70a 100755
--- a/transcrypt
+++ b/transcrypt
@@ -591,7 +591,8 @@ save_configuration() {
 
 	# write the filter settings. Sorry for the horrific quote escaping below...
 	# shellcheck disable=SC2016
-	transcrypt_path='"$(git config transcrypt.crypt-dir 2>/dev/null || printf %s/crypt ""$(git rev-parse --git-dir)"")"/transcrypt'
+	# transcrypt_path='"$(git config transcrypt.crypt-dir 2>/dev/null || printf %s/crypt ""$(git rev-parse --git-dir)"")"/transcrypt'
+	transcrypt_path=$(realpath "$0" 2>/dev/null)
 
 	# Ensure filter attributes are always set for the default (unspecified) context
 	git config filter.crypt.clean "$transcrypt_path clean context=default %f"
-- 
2.38.1

